optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/KieBaseUtil.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.director.drools;
+
+import org.drools.ancompiler.KieBaseUpdaterANC;
+import org.drools.core.base.CoreComponentsBuilder;
+import org.kie.api.KieBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class KieBaseUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KieBaseUtil.class);
+    private static final String ANC_ENABLED_PROPERTY = "optaplanner.drools.compile_alpha_network";
```

Comment:
Let's not use a system property, but a field in ScoreDirectorFactoryConfig
That what it's consistent with SolverConfig.domainAccessType
and we can run multiple solver instances with a different setting in the same JVM.

Reply:
It will also need fail fasts:
if the score director is not CS-D, then fail fast if that field is non-null.